local/code/game/objects/items/neuroware/_neuroware.dm

```diff
@@ -0,0 +1,189 @@
+	screwdriver.play_tool_sound(src, 100)
+	if(!screwdriver.use_tool(src, user, 2 SECONDS))
+		balloon_alert(user, "interrupted!")
+		return ITEM_INTERACT_BLOCKING
+
+	can_overdose = !can_overdose
+
+	playsound(src, 'sound/machines/click.ogg', 50, TRUE, SILENCED_SOUND_EXTRARANGE)
+	balloon_alert(user, "safety [can_overdose ? "disabled" : "enabled"]")
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/disk/neuroware/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
+	. = ..()
+	if(can_hack && (uses > 0) && istype(held_item, /obj/item/screwdriver))
+		context[SCREENTIP_CONTEXT_LMB] = "[can_overdose ? "Enable" : "Disable"] overload safety"
+		return CONTEXTUAL_SCREENTIP_SET
+
+/obj/item/disk/neuroware/attack_self(mob/user, modifiers)
+	if(!try_install(user, user))
+		return ..()
+
+/obj/item/disk/neuroware/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!isliving(interacting_with))
+		return NONE
+	return try_install(interacting_with, user) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
+
+
+///Safely implement any side-effects after installing.
+/obj/item/disk/neuroware/proc/after_install(mob/living/carbon/human/target, mob/living/carbon/human/user)
+	return
+
+///Returns TRUE if overdose would occur upon install(), overwise returns FALSE.
+/obj/item/disk/neuroware/proc/check_overdose(mob/living/carbon/human/target, list/reagent_list)
+	for(var/reagent_type as anything in reagent_list)
+		var/datum/reagent/existing_reagent = target.has_reagent(reagent_type)
+		if(!existing_reagent || existing_reagent.overdose_threshold == 0)
+			continue
+		var/new_volume = reagent_list[reagent_type] + existing_reagent.volume
+		if(new_volume >= existing_reagent.overdose_threshold)
+			return TRUE
+	return FALSE
+
+///Installs only if the mob has a synthetic brain, unless they got a nif
+/obj/item/disk/neuroware/proc/try_install(mob/living/carbon/human/target, mob/living/carbon/human/user)
+	if(!ishuman(target))
+		return
+	if(uses == 0)
+		balloon_alert(user, "it's been used up!")
+		return
+	if(target != user)
+		target.visible_message(
+			span_danger("[user] tries to force [src] into [target]'s [NEURO_SLOT_NAME]!"),
+			span_userdanger("[user] tries to force [src] into your [NEURO_SLOT_NAME]!")
+		)
+		if(target.is_blind())
+			to_chat(target, span_userdanger("You feel something being inserted into your [NEURO_SLOT_NAME]!"))
+		if(external_delay > 0)
+			user.balloon_alert_to_viewers("inserting chip...")
+			if(!do_after(user, 5 SECONDS, target))
+				return
+		target.visible_message(
+			span_danger("[user] forces [src] into [target]'s [NEURO_SLOT_NAME]!"),
+			span_userdanger("[user] forces [src] into your [NEURO_SLOT_NAME]!")
+		)
+		if(target.is_blind())
+			to_chat(target, span_userdanger("Something was inserted into your [NEURO_SLOT_NAME]!"))
+
+	// Prevent reagent overdose if safety is enabled
+	if(length(list_reagents) && !can_overdose && check_overdose(target, list_reagents))
```

Comment:
would it make more sense to drop the length check and move the `can_overdose` check to `check_overdose`? If the list is empty the loop will end and return false, and it makes more sense that the caller doesn't have to worry about checking `can_overdose`, that's the responsibility of `check_overdose`